apps/src/music/blockly/MusicBlocklyWorkspace.js

```diff
@@ -196,12 +198,14 @@ export default class MusicBlocklyWorkspace {
         [
           BlockTypes.TRIGGERED_AT,
           BlockTypes.TRIGGERED_AT_SIMPLE,
+          BlockTypes.TRIGGERED_AT_SIMPLE2,
           BlockTypes.NEW_TRACK_ON_TRIGGER
         ].includes(block.type)
       ) {
         const id = block.getFieldValue(TRIGGER_FIELD);
         events[this.triggerIdToEvent(id)] = {
-          code: Blockly.JavaScript.blockToCode(block)
+          code:
+            Blockly.JavaScript.blockToCode(block) + functionImplementationsCode
```

Comment:
To clarify...is this attaching all `functionImplementationsCode` to each trigger? Won't this trigger all functions on every trigger?

Reply:
It attaches the implementations of the functions, but they won't be called unless some code in the trigger handler explicitly calls one of them.

Reply:
Ahh right, okay got it.